src/lib.rs

```diff
@@ -626,6 +626,17 @@ arbitrary_array! { 32, (T, a) (T, b) (T, c) (T, d) (T, e) (T, f) (T, g) (T, h)
 (T, z) (T, aa) (T, ab) (T, ac) (T, ad) (T, ae) (T, af)
 (T, ag) }
 
+impl<'a> Arbitrary<'a> for &'a [u8] {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        u.get_bytes(u.len())
```

Comment:
Consuming the entire unstructured buffer every time would mean that e.g.

```rust
#[derive(Arbitrary)]
struct Foo<'a> {
    bytes: &'a [u8],
    other_bytes: &'a [u9],
}
```

would always generate an empty slice for the 2nd field. You may want to do the `arbitrary_len` thing still here.

Reply:
Good eye @nagisa. Yes, we should still do the `arbitrary_len` thing, but also implement `arbitrary_take_rest` with this implementation that takes all the bytes.

Reply:
> Consuming the entire unstructured buffer every time would mean that e.g.
> 
> ```rust
> #[derive(Arbitrary)]
> struct Foo<'a> {
>     bytes: &'a [u8],
>     other_bytes: &'a [u9],
> }
> ```
> 
> would always generate an empty slice for the 2nd field. You may want to do the `arbitrary_len` thing still here.

great catch! fixed in https://github.com/rust-fuzz/arbitrary/pull/67/commits/f951b5abbcafb8235670aa1afe426b535cdba81f

> Good eye @nagisa. Yes, we should still do the `arbitrary_len` thing, but also implement `arbitrary_take_rest` with this implementation that takes all the bytes.

the default implementation for `arbitrary_take_rest` defers to `arbitrary`:

https://github.com/rust-fuzz/arbitrary/blob/f951b5abbcafb8235670aa1afe426b535cdba81f/src/lib.rs#L173-L180

so seems like it'll already use this code. is that what you were thinking?

Reply:
Not quite. It should be this:

```rust
impl<'a> Arbitrary<'a> for &'a [u8] {
    // ...
    fn arbitrary_take_rest(mut u: Unstructured<'a>) -> Result<Self> {
        Ok(u.take_rest())
    }
}
```

This will always consume the rest of the `Unstructured`'s input (which is what we want in the `_take_rest` method) without getting any kind of length prefix (which we want in regular `arbitrary`, but *not* in `arbitrary_take_rest`).

Does that clear things up? Sorry this required so many back and forths to spell out